packages/flutter/lib/src/widgets/app.dart

```diff
@@ -231,11 +290,25 @@ class _WidgetsAppState extends State<WidgetsApp> implements WidgetsBindingObserv
   GlobalObjectKey<NavigatorState> _navigator;
   Locale _locale;
 
+  Locale _resolveLocale(Locale newLocale, Iterable<Locale> supportedLocales) {
+    if (widget.resolveLocaleCallback != null)
+      return widget.resolveLocaleCallback(newLocale, widget.supportedLocales);
```

Comment:
it might make sense to only return here if the return value was non-null. that would allow people to use null as a "do the right thing" fallback.

Reply:
Good idea.